### PR TITLE
Call abort in place of Cancel if test session timeout reach

### PR DESCRIPTION
**Pull request by Faizan2304**
**Issue1:** Suppose there are two test in one project, first one takes 2 second and second one takes 2 minutes and we are using TestSessionTimeout=5000 (5 second). vstest.console.exe runs both test.

Expected: Only first test should run and abort should get called.
Actual: vstest,console is sending cancel request to adapter. Adapter runs second test and then abort run.

**Issue2:**: https://github.com/Microsoft/vstest/issues/1077

### CODE REVIEW

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ internal class ParallelProxyExecutionManager : ParallelOperationManager<IProxyEx
     {
         #region TestRunSpecificData
 
+        private bool abortRequested = false;
```
Comment (codito) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs]: This code requires comments on how the `explicit` and `implicit` aborts differ.

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs [hunk 2]
```diff
@@ -499,6 +507,7 @@ public void TestRunCompleteHandlerShouldNotWriteTolatTestToConsoleIfTestsAborted
             this.mockOutput.Verify(o => o.WriteLine(CommandLineResources.TestRunAborted, OutputLevel.Error), Times.Once());
         }
 
+
```
Comment (codito) [inline test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: remove

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs [hunk 3]
```diff
@@ -514,6 +523,14 @@ public void TestRunCompleteHandlerShouldWriteToConsoleIfTestsAborted()
         }
 
         [TestMethod]
+        public void TestRunCompleteHandlerShouldWriteToConsoleIfTestsAbortedWithoutRunningAnyTest()
```
Comment (codito) [inline test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: Should we add tests for this change?
https://github.com/Microsoft/vstest/pull/1133/files#diff-f550c5f97286ac7d288d9c83547cff40R451
https://github.com/Microsoft/vstest/pull/1133/files#diff-f550c5f97286ac7d288d9c83547cff40R455

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs (hunk 1 above)
Comment (Faizan2304) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs]: Fixed!!

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs (hunk 2 above)
Comment (Faizan2304) [reply test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: Fixed

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs (hunk 3 above)
Comment (Faizan2304) [reply test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: We already have test for all the above mention scenario.
https://github.com/Microsoft/vstest/pull/1133/files#diff-1f79d9ff9499d17db1e9f39c5d479602R447
https://github.com/Microsoft/vstest/pull/1133/files#diff-1f79d9ff9499d17db1e9f39c5d479602R447
https://github.com/Microsoft/vstest/pull/1133/files#diff-1f79d9ff9499d17db1e9f39c5d479602R534

**Pull request merged**